### ISSUE TITLE
Extend CylMesh to handle hollow cylinders

### ISSUE THREAD

**Issue by marrov** [simpeg/discretize#210]
As a follow-up to the question I asked on Slack (Simpeg#discretize), I'd like to suggest to extend the functionality of the function CylMesh to be able to handle hollow cylinders.

Currently, the only way to discretize the volume of an annular reactor (i.e. hollow cylinder) is by creating two cylinder meshes, one with r=r_in and the other with r=r_out. Then, field values for each cell are computed by subtracting the inner mesh values from the outer mesh values. A less computationally intensive approach is to allow CylMesh to create hollow cylinders. 

As suggested on Slack, using the offset argument of the CylMesh function to shift the starting radius by changing the x0 property of the mesh (e.g. setting x0=r_in) does not work. This happens because the the initial x-node (radial node) location is hard-coded to zero.

A minimal working example (MWE.py) that showcases this behaviour can be found below:
```python
from discretize import CylMesh
import numpy as np

# Annular reactor geometry 
L = 0.28                # Lamp length [m]
r_inner = 0.0115        # Quartz sleeve radius [m]
r_outer = 0.0425        # Reactor radius [m]
r = r_outer-r_inner     # Radial extent of the reactor [m]

# Compute the true reactor volume
V_reactor = np.pi*L*(r_outer**2-r_inner**2)
print(f'Reactor volume \t\t\t\t\t\t\t\t = {V_reactor:.5f} \t m**2')

# Define mesh parameters that remain the same despite the approach
ncr = 50                # number of mesh cells in r
ncp = 50                # number of mesh cells in phi
ncz = 50                # number of mesh cells in z
dz = L/ncz              
hp = (2*np.pi/ncp)*np.ones(ncp)
hz = dz*np.ones(ncz)

# Approach A: using two meshes - one for the inner and one for the outer cyl
dr_outer = r_outer/ncr
dr_inner = r_inner/ncr
hr_outer = dr_outer*np.ones(ncr)
hr_inner = dr_inner*np.ones(ncr)
offset = [0, 0, -L/2]
mesh_outer = CylMesh([hr_outer, hp, hz], offset)
mesh_inner = CylMesh([hr_inner, hp, hz], offset)
V_discretized_A = np.sum(mesh_outer.vol) - np.sum(mesh_inner.vol)
print(f'Discretized reactor volume by approach A \t = {V_discretized_A:.5f} \t m**2')

# Approach B: using an offset and one single mesh
dr = r/ncr
hr = dr*np.ones(ncr)
offset = [r_inner, 0, -L/2] # Also tried both x0, y0 = r_inner, r_inner
mesh = CylMesh([hr, hp, hz], offset)
cc = mesh.gridCC
V_discretized_B = np.sum(mesh.vol)
print(f'Discretized reactor volume by approach B \t = {V_discretized_B:.5f} \t m**2')
```

**Comment by lheagy** [comment]
Thanks @mrovirasacie for raising this! Very cool to see that you are interested in using hollow cylinders - this isn't something we have played around with much (as you have found out!). There are a few places where we have baked-in the assumption that the initial x-node (radial node) location is at zero. e.g. 

https://github.com/simpeg/discretize/blob/9c132095e50daddf1e4debac89df4ebf3be87924/discretize/CylMesh.py#L355-L360

https://github.com/simpeg/discretize/blob/9c132095e50daddf1e4debac89df4ebf3be87924/discretize/CylMesh.py#L343-L346
